### ISSUE TITLE
Add Prettier config file

### ISSUE THREAD

**Issue by netbek** [emeeks/semiotic#153]
With the ESLint rules now in place (https://github.com/emeeks/semiotic/issues/151), I think enabling consistent code formatting with [Prettier](https://prettier.io/) could be a win.

Prettier could be added to the workflow in various ways:
* [CLI](https://prettier.io/docs/en/cli.html)
* IDE plugin (format on save), e.g. for [Atom](https://github.com/prettier/prettier-atom)
* [Pre-commit hook](https://prettier.io/docs/en/precommit.html)

ESLint and Prettier can be configured to play nicely with each other.

Without having to investigate aforementioned, it would be lovely to have a config file in the repo, to start. Here's one I put together based on the ESLint rules: https://github.com/emeeks/semiotic/pull/152

I've tested it by doing the following:

```
npm i -g prettier
prettier --write "src/components/**/*.js"
npm test
```

All tests pass, and output from `git diff` shows only 7 files were changed, minimally.

**Comment by emeeks** [comment]
I already use prettier, I just use it across my projects and don't specify it for Semiotic. I don't like to be opinionated about how people format their code but if you think it would be helpful I can expose the prettier settings I use in a `.prettierrc` like you have in the PR.

**Comment by netbek** [comment]
Sorry for the delayed reply. My motivation for including a Prettier config is so that code can be auto-formatted in a manner that's consistent with the linter config. I assume the intention behind including a linter config was to communicate code style conventions. I prefer Prettier to `eslint --fix` for formatting because Prettier does more, and unlike other beautifying plugins I've tried, it outputs code that agrees with the linter config 99% of the time (array elements are sometimes indented a bit too much according to the linter). In short, I use it so that I don't have to think about the linter complaining, and because I have a strong preference for consistently formatted code. Incidentally, I've configured my IDE to only auto-format with Prettier if a `.prettierrc` file exists in the root of a project/folder, so that projects can have different configs and some can be ignored by Prettier entirely.

As I see it, there'd be no harm in including a config for people who use Prettier, and those who don't won't be negatively affected. Your call, though :-) It might be helpful to potential contributors if the contributing guidelines noted the maintainer's preferred code style (or lack thereof).
